### PR TITLE
Add test for ensuring hunks can lock across branches

### PR DESCRIPTION
**Pull request by mtsgrd**
While we haven't yet decided how to handle this edge case we still want to test that locking works as expected. With two hunks committed into separate branches, introducing a third that touches both creates a situation where it's locked to both.

### CODE REVIEW

File: crates/gitbutler-core/tests/suite/virtual_branches/create_commit.rs [hunk 1]
```diff
@@ -221,43 +182,167 @@ async fn should_double_lock() {
...
+    let files = get_virtual_branch(controller, project_id, branch_2_id)
+        .await
+        .files;
+    assert_eq!(files.len(), 1);
+}
+
+// TODO: Where should these helper functions live?
+
+fn write_file(repository: &TestProject, path: &str, lines: &Vec<String>) {
+    fs::write(repository.path().join(path), lines.join("\n")).unwrap()
```
Comment (mtsgrd) [inline crates/gitbutler-core/tests/suite/virtual_branches/create_commit.rs:275]: There are 197 other variations of this line where this could be used, but we'd need a function that accepts lines as a vector or as a single string.

File: crates/gitbutler-core/tests/suite/virtual_branches/create_commit.rs [hunk 2]
```diff
@@ -221,43 +182,167 @@ async fn should_double_lock() {
...
+    branch_id: Id<Branch>,
+) -> VirtualBranch {
+    controller
+        .list_virtual_branches(project_id)
+        .await
+        .unwrap()
+        .0
+        .into_iter()
+        .find(|b| b.id == branch_id)
+        .unwrap()
```
Comment (mtsgrd) [inline crates/gitbutler-core/tests/suite/virtual_branches/create_commit.rs:301]: @krlvi I see you have a `get_branch` function in `VirtualBranchesHandle`. Could we make tests use this struct as well?

File: crates/gitbutler-core/tests/suite/virtual_branches/create_commit.rs [hunk 3]
```diff
@@ -221,43 +182,185 @@ async fn should_double_lock() {
...
+        .create_virtual_branch(project_id, &branch::BranchCreateRequest::default())
+        .await
+        .unwrap();
+
+    let mut request_opts = branch::BranchCreateRequest::default();
+    request_opts.selected_for_changes = Some(true);
+    let branch_2_id = controller
+        .create_virtual_branch(project_id, &request_opts)
+        .await
+        .unwrap();
```
Comment (mtsgrd) [inline crates/gitbutler-core/tests/suite/virtual_branches/create_commit.rs]: Is there a way of inlining the `selected_for_changes` option?
